### ISSUE TITLE
why not expand it into the view! macro

### ISSUE THREAD

**Issue by gdennie** [blorbb/leptos-mview#2]
Since `view!` already possess a lot of optimizations; as an initial effort, why not expand `mview!` into `view!` instead of directly into the builder pattern. Subsequently, as your more concise syntax catches the core effort around `view!` might well spread to `mview!`. After all, the added value is in the concise syntax and not its expansion into the UI builder pattern. Nonetheless, looking good.

**Comment by blorbb** [comment]
the builder pattern gives more control than expanding to the `leptos::view!` macro. I would like to support expanding to the leptos view eventually, just haven't had the time to work on it yet.

Plans for the new renderer in leptos 0.6 also completely changes how `view!` works, and it supposedly means that `view!` will always expand to the same thing, whether on the client or server. When it's released, I can probably use the new renderer directly without any performance downsides.

However, expanding to the `leptos::view!` macro may also have some issues.

The leptos macro also lacks a few features that this one has: the main one being patterns in `let:` for children.
```rust
let letters = ['a', 'b', 'c'];
view! {
    For
        each=[letters.into_iter().enumerate()]
        key={|(i, _)| *i}
    |(i, letter)| {
        "letter " {i+1} " is " {letter}
    }
};
```
In leptos, you can only pass a single ident to `let:`, which means each segment has to be accessed like `item.0`, `item.1`
```rust
leptos::view! {
    <For each=move || letters.into_iter().enumerate() key=|(i, _)| *i let:item>
        "letter " {item.0 + 1} " is " {item.1}
    </For>
}
```

Another thing that will probably be a problem is the poorer rust-analyzer support. Since leptos supports any expressions after an attribute (like `key=|(i, _)| *i` above), it needs to parse a `syn::Expr` to know when the value stops and the next attribute begins (`let:item` immediately after). But this means that invalid expressions are not parsed and rust-analyzer can't help (e.g. `each=letters.` would not provide autocomplete suggestions after the `.`)
(my macro gets around this by requiring braces for non-literal expressions, so that everything inside is just parsed as a `TokenStream` and always successfully expands, even if the syntax is invalid)

I may add this feature in the future (perhaps with a feature flag to opt-in), though 0.6 may make this obsolete. I'll keep this issue open in case its needed though.

**Comment by gdennie** [comment]
Thank you very much for taking the time to deliver a comprehensive and informative response. :+1:

**Comment by blorbb** [comment]
closed as no longer necessary due to leptos 0.7 being released :)
